src/tests/test_integration.py

```diff
@@ -80,9 +80,10 @@ def invoke(self, args="", target=None):
         run_target = self.tempdir if target is None else \
             os.path.join(self.tempdir, target)
 
-        cmd = shlex.split("{} {} {}"
-                          .format(self.script_name, run_target, args),
-                          posix=False)
+        cmd = shlex.split(
+            ('"{}" -m {} {} {}' if sys.platform == "nt" else '{} -m {} {} {}')
```

Comment:
Actually, this line was also testing that `pydocstyle` is installed as an executable by itself - there's no reason to use `sys.executable` here.

The change below in `test_run_as_named_module` is great though.

Reply:
As this is a reusable component, can it be using the more reliable and faster CLI entry point `__main__` ?  This then covers the same paths as `test_run_as_named_module`

And then instead of `test_run_as_named_module` I create some dedicated tests for the generated stub created to invoke the entry point.

Then distro packagers can disable the tests for the generated stub, because e.g. atm openSUSE only creates the stub entry point for python3 , and therefore the py2 integration tests fail.

Reply:
I don't mind making `invoke` use the `- m` way and changing `test_run_as_named_module` to, e.g., `test_run_as_script` if that's what you're suggesting. As long as they are both being tested.

Reply:
Yup; that is what I had in mind.  I'll revise the PR over the w/e to that effect.

Reply:
Ping, still working on this?